docs/configdoc.md

```diff
@@ -0,0 +1,117 @@
+environment: PROD
+....
+```
+
+### Below is the detailed documentation for each field
+
+
+## [l3afd]
+| FieldName     | Example       | Description     |
+| ------------- | ------------- | --------------- |
+|pid-file| `"l3afd.pid"`  |It contains process id of eBPF Programs |
+|datacenter| `"dummy"` | Name of Datacenter|
+|bpf-dir| `"/dev/shm"` | Absolute Path of eBPF Programs (where they are extracted) |
+|bpf-log-dir|`""`      | Absolute Path,where we store logs eBPF Programs|
+|kernel-major-version|`"4"`|Major version of the kernel|
+|kernel-minor-version|`"15"`|Miner version of the kernel (Ex 4.15)|
+|shutdown-timeout|`"1s"`|It is the amount of time allowed to stop all network functions|
+|http-client-timeout|`"10s"`|It is the amount of time allowed to get response headers|
+|max-nf-restart-count|`"3"`|Maximum number of attempts to monitor eBPF Program|
+|bpf-chaining-enabled|`"true"`|Boolean to set bpf-chaining |
+|swagger-api-enabled|`"false"`|Boolean to check swagger api enable or not|
+|environment|`"PROD"`| Environment  variable |
+|BpfMapDefaultPath|`"/sys/fs/bpf"`|It is a base path for storing bpf maps|
+|TcMapsRelativePath|`"tc/globals"`|It is a relative path of tc maps with respect to BpfMapDefaultPath|
+## [kf-repo]
+| FieldName     | Example       | Description     |
+| ------------- | ------------- | --------------- |
+|url| `"http://localhost:8000/"`|Default Repository to Download eBPF Programs|
+
+## [web]
+| FieldName     | Example       | Description     |
+| ------------- | ------------- | --------------- |
+|metrics-addr|`"0.0.0.0:8898"`|Prometheus endpoint for pull/scrape the metrics|
+|kf-poll-interval|`"30s"`|Periodic Interval to scrape metrics using Prometheus|
+|n-metric-samples|`"20"`|Number of Metric Samples|
+
+## [admind]
+| FieldName     | Example       | Description     |
+| ------------- | ------------- | --------------- |
+|host|`""`|Hostname of admin|
+|username|`""`|Username of admin|
+|api-key|`""`|Api key of admin|
+|groud-id|`"0"`|Group id of admin|
+|api-enabled|`"true"`|Boolean to check api enabled or not|
+
+## [xdp-root-program]
+| FieldName     | Example       | Description     |
+| ------------- | ------------- | --------------- |
+|name|`"xdp-root"`|Name of xdp-root program|
+|artifact|`"xdp-root.tar.gz"`|Archive of xdp-root program|
+|ingress-map-name|`"root_array"`|Ingress map name of xdp-root program|
+|command|`"xdp_root"`|Command to run xdp-root program|
+|version|`"1.01"`|Version of xdp-root program|
+|is-user-program|`"false"`|Boolean to check xdp-root is kernel space or user space program|
+
+## [tc-root-program]
+| FieldName     | Example       | Description     |
+| ------------- | ------------- | --------------- |
+|name|`"tc_root"`|Name of tc_root program|
+|artifact|`"l3af_tc_root.tar.gz"`|Archive of tc_root program|
+|ingress-map-name|`"tc_ingress_root_array"`|Ingress map name of tc_root program|
+|egress-map-name|`"tc_egress_root_array"`|Egress map name of tc_root program|
```

Comment:
What is an "egress map"?   Maybe provide a link to other documentation about the tc-root program requirements?